src/server/server.go

```diff
@@ -63,6 +69,32 @@ func PprofMem(w http.ResponseWriter, r *http.Request) {
         }
 }
 
+func PprofCpuStart(w http.ResponseWriter, r *http.Request) {
+	log.Printf("Starting cpu profiling\n")
+	w.Header().Add("Content-Type", "application/octet-stream")
+
+	if err := pprof.StartCPUProfile(w); err != nil {
+	    log.Fatal("could not start CPU profile: ", err)
+	}
+
+	select {
+	case <-cpuProfTimeout:
+		break
+	case <-time.After((1 << 63 - 1) * time.Nanosecond): // max duration
```

Comment:
Is there value in having a 100+ year timeout?

Reply:
I meant to write a wait-and-interrupt mechanism. I chose a 100+ year timeout (actually, the max duration the library supports) to simulate the "infinite wait" until an "interrupt" (`cpuProfTimeout` channel gets filled). [Reference](https://stackoverflow.com/a/23331563/11247758)

Reply:
If it's infinite wait, then you don't need the select/case.  It's just a one-liner:

<-cpuProfTimeout

But I'm hoping we can get rid of the channel altogether.